test/_spy_method.js

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and other contributors where applicable.
+ * Licensed under the BSD 2-Clause License; you may not use this file except in
+ * compliance with the BSD 2-Clause License.
+ */
+
+'use strict';
+
+const kOrigFunction = Symbol('kOrigFunction');
+
+class SpyObject {
+  constructor() {
+    this.calls = [];
+  }
+
+  /**
+   * Tells if the spied function has been called. If a positive number
+   * is passed checks if it's equal to the times it has been called
+   *
+   * @param {Number} [times=1] - To check for specific number of calls
+   * @returns {Boolean}
+   */
+  hasBeenCalled(times) {
+    const t = typeof times === 'number' ? times : 1;
+
+    return this.calls.length === t;
+  }
+}
+
+/**
+ * Returns an object with data about the usage of the spied method
+ *
+ * @param {any} test the test where the spy is created
+ * @param {Object} target the object which holds the method to spy
+ * @param {String} method the method to spy on
+ * @returns {SpyObject} object with usage data
+ */
+function spyOn(test, target, method) {
```

Comment:
nit: I would rename this `test` to `t`, which is the typical var name used in tape and tap tests.

Reply:
finally I decided to remove `spyOn` method since it's not necessary in this tests